cli/src/semdep/parsers/util.py

```diff
@@ -330,3 +333,38 @@ def lexeme(p: "Parser[A]") -> "Parser[A]":
     ),
 )
 json_doc = whitespace >> json_value
+
+
+# This is truly awful. Python's recursion limit is too small for some realistic package-lock.json files.
+# So we manually patch the parse and parse_partial methods of the json parser to increase it 😵‍💫
```

Comment:
why not just change the limit for the whole duration of execution?

Reply:
good point, done